Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=26799a63110dcbe81291ea53178f6b4810d07424
Commit:     26799a63110dcbe81291ea53178f6b4810d07424
Parent:     b58602a4bac012b5f4fc12fe6b46ab237b610d5d
Author:     Ravikiran G Thirumalai <kiran@scalex86.org>
AuthorDate: Wed Dec 31 13:44:46 2008 -0800
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Fri Jan 2 17:16:29 2009 +0100

    x86: fix incorrect __read_mostly on _boot_cpu_pda
    
    The pda rework (commit 3461b0af025251bbc6b3d56c821c6ac2de6f7209)
    to remove static boot cpu pdas introduced a performance bug.
    
    _boot_cpu_pda is the actual pda used by the boot cpu and is definitely
    not "__read_mostly" and ended up polluting the read mostly section with
    writes.  This bug caused regression of about 8-10% on certain syscall
    intensive workloads.
    
    Signed-off-by: Ravikiran Thirumalai <kiran@scalex86.org>
    Acked-by: Mike Travis <travis@sgi.com>
    Cc: <stable@kernel.org>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

 arch/x86/kernel/head64.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/x86/kernel/head64.c b/arch/x86/kernel/head64.c
index 388e05a..b9a4d8c 100644
--- a/arch/x86/kernel/head64.c
+++ b/arch/x86/kernel/head64.c
@@ -27,7 +27,7 @@
 #include <asm/trampoline.h>
 
 /* boot cpu pda */
-static struct x8664_pda _boot_cpu_pda __read_mostly;
+static struct x8664_pda _boot_cpu_pda;
 
 #ifdef CONFIG_SMP
 /*
